core/cli/src/index.ts

```diff
@@ -27,6 +36,23 @@ export const shouldDisableNativeFetch = (): boolean => {
   )
 }
 
+const loadTasks = async (logger: Logger, config: ValidConfig): Promise<Validated<Record<string, Task>>> =>
+  mapValidated(
+    reduceValidated(
+      await Promise.all(
+        Object.entries(config.tasks).map(async ([taskName, pluginId]) =>
+          flatMapValidated(await importPlugin(pluginId), (plugin) =>
+            mapValidated(validatePluginTasks(plugin as RawPluginModule), (tasks) => [
+              taskName,
+              new tasks[taskName](logger, taskName, getOptions(pluginId as OptionKey) ?? {})
+            ])
+          )
+        )
+      )
+    ),
+    Object.fromEntries
+  )
+
```

Comment:
**suggestion:** these calls are very nested :O is there a way to split this into steps with named variables? I think it would be a lot easier to read